intake_dal/dal_online.py

```diff
@@ -72,9 +73,12 @@ def get_metadata(key: str, default):
 
     def _get_partition(self, _) -> pd.DataFrame:
         self._get_schema()
-        return deserialize_avro_str_to_pandas(
-            _http_get_avro_data_set(self._url, self._canonical_name, self._key_value), self._avro_schema
-        )
+        data = _http_get_avro_data_set(self._url, self._canonical_name, self._key_value)
+        for item in data:
+            for key, field in item.items():
+                if isinstance(field, dict) and "DATETIME" in field:
+                    item[key] = datetime.strptime(field["time"])
+        return pd.DataFrame(data)
```

Comment:
A nice to have is a unit test that measures the performance of JSON -> pandas to ensure it doesn't regress, or at the least print out the performance.